src/ReturnTypes/CollectionFilterRejectDynamicReturnTypeExtension.php

```diff
@@ -51,10 +53,16 @@ public function getTypeFromMethodCall(
             return null;
         }
 
+        $methodName = $methodReflection->getName();
+        assert($methodName === 'filter' || $methodName === 'reject', 'proven in isMethodSupported');
+
         if (count($methodCall->getArgs()) < 1) {
-            $nonFalseyTypes = TypeCombinator::removeFalsey($valueType);
+            $modifiedType = match ($methodName) {
+                'filter' => TypeCombinator::removeFalsey($valueType),
+                'reject' => TypeCombinator::intersect($valueType, StaticTypeFactory::falsey())
```

Comment:
Fails with the following error:

```
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   ReturnTypes/CollectionFilterRejectDynamicReturnTypeExtension.php
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------
  62     Calling PHPStan\Type\StaticTypeFactory::falsey() is not covered by backward compatibility promise. The method might change in a minor PHPStan version.
         💡 If you think it should be covered by backward compatibility promise, open a discussion:
         💡 https://github.com/phpstan/phpstan/discussions
         💡
         💡 See also:
         💡 https://phpstan.org/developing-extensions/backward-compatibility-promise

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------
```

How should we handle this? I found no other way to do this, `TypeCombinator` does not offer something like `removeTruthy`.

Reply:
Can you open a discussion in the PHPStan repo, like the error message says? I think it is a use case and we can add `@api` annotation to the method. Or like you said maybe we can introduce `removeTruthy` method.

Reply:
I opened https://github.com/phpstan/phpstan/discussions/12039. Are you fine with ignoring the error for now? Since tests run on both the highest and lowest allowed dependencies, we should notice when the API breaks.

Reply:
We can now use `TypeCombinator::removeTruthy` We also need to update PHPStan version in composer.json to latest one

Reply:
Thanks for helping to land this feature in PHPStan @canvural, I updated to use it.